sycl/plugins/cuda/pi_cuda.cpp

```diff
@@ -567,30 +567,6 @@ pi_result _pi_program::build_program(const char *build_options) {
   return success ? PI_SUCCESS : PI_BUILD_PROGRAM_FAILURE;
 }
 
-/// Finds kernel names by searching for entry points in the PTX source, as the
-/// CUDA driver API doesn't expose an operation for this.
-/// Note: This is currently only being used by the SYCL program class for the
-///       has_kernel method, so an alternative would be to move the has_kernel
-///       query to PI and use cuModuleGetFunction to check for a kernel.
-/// Note: Another alternative is to add kernel names as metadata, like with
-///       reqd_work_group_size.
-std::string getKernelNames(pi_program program) {
```

Comment:
maybe DIE_NOIMPLEMENTATION?
Listing kernels will likely be needed for dynamic device code linkage. @AlexeySachkov - FYI

Reply:
Good point. I'll make the change.